### PR TITLE
Fix for bug #1457 in convolution operator with dilate > 2

### PR DESCRIPTION
**Pull request by kadeng**
Fix for https://github.com/dmlc/mxnet/issues/1457 

I.e. this fixes an issue in the convolution operator if the dilate parameter is set to a value larger than 2.

This is the second part of the fix. The first part went into mshadow, and has already been merged. See

https://github.com/dmlc/mshadow/pull/102

Now it is neccessary to apply this, or the convolution op will be completely broken for dilate param > 2

### CODE REVIEW

File: tests/python/unittest/test_convolution_dilate.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    be = net.bind(mx.cpu(), args={ 'input' : white_in, 'test_convolution_weight' : dkernel})
+
+    be.forward(True)
+    out = be.outputs[0].asnumpy()
+    # Now do a simple check of the kernel gradient
+    assert(out[0,0,16,16] - np.sum(kernel_gradient) - out_orig[0,0,16,16] < 0.001)
+    
+
+if __name__=='__main__':  
+	test_impulse_response((1,1), kernel_shape=(3,3), verbose=False)
```
Comment (piiswrong) [inline tests/python/unittest/test_convolution_dilate.py]: condense these with a loop and move it to test_operators.py

**Comment by kadeng** [comment]
Moved the test, as requested..

**Comment by tqchen** [comment]
@kadeng can you test locally why the test_operators failed?

**Comment by kadeng** [comment]
@tqchen:

I could determine two problems:

1.) My unit test had some python 2 style print statements. I have just removed them, since they could trigger syntax errors with python 3.

2.) If I build the current, unmodified master, and then run test_operators.py, I already get an exception that seems to be completely unrelated: 

```
Traceback (most recent call last):
  File "test_operator.py", line 739, in <module>
    test_slice_channel()
  File "test_operator.py", line 171, in test_slice_channel
    check_slice_channel(2, 4)
  File "test_operator.py", line 73, in check_slice_channel
    op = mx.sym.SliceChannel(data=data, num_outputs=shape[1], squeeze_axis=1)
  File "/home/kai/Projekte/mxnet/mxnet/python/mxnet/symbol.py", line 988, in creator
    ctypes.byref(sym_handle)))
  File "/home/kai/Projekte/mxnet/mxnet/python/mxnet/base.py", line 77, in check_call
    raise MXNetError(py_str(_LIB.MXGetLastError()))
mxnet.base.MXNetError: Cannot find argument 'squeeze_axis', Possible Arguments:
----------------
num_outputs : int, required
    Number of outputs to be sliced.
axis : int, optional, default='1'
    Dimension along which to slice.

```

**Comment by kadeng** [comment]
Found the problem, had to do with me not being used to working with git submodules until now. All checks pass now, and it should be safe to merge..

**Pull request merged**